Mathlib/LinearAlgebra/Matrix/SVD/Defs.lean

```diff
@@ -0,0 +1,479 @@
+    simp only [ne_eq, Equiv.symm_trans_apply, Equiv.symm_symm, Equiv.coe_fn_symm_mk, Sum.elim_inr,
+      Equiv.sumCompl_apply_inr]
+    simp_rw [h, diagonal_apply_eq]
+    apply eigen_eigenRowEquiv_inr
+  · simp only [diagonal_apply_ne, ne_eq, EmbeddingLike.apply_eq_iff_eq, Sum.inr.injEq, h]
+
+lemma S_σpos_block (A : Matrix (Fin M) (Fin N) 𝕂) :
+    A.svdS.toBlocks₁₁ = A.svdμ := by
+  unfold toBlocks₁₁ svdμ svdS
+  simp only [reindex_apply, submatrix_apply, ne_eq, EmbeddingLike.apply_eq_iff_eq, Sum.inl.injEq,
+    submatrix_diagonal_equiv]
+  funext i j
+  by_cases h: i=j
+  · simp_rw [h]
+    unfold eigenColumnEquiv finRankEquivEigsConjTransposeMulSelf
+      Equiv.sumCongr Sum.map
+    simp only [ne_eq, Equiv.symm_trans_apply, Equiv.symm_symm, Equiv.coe_fn_symm_mk,
+      Sum.elim_inl, Equiv.sumCompl_apply_inl, of_apply,
+      diagonal_apply_eq, Function.comp_apply]
+  · rw [diagonal_apply_ne, of_apply, diagonal_apply_ne]
+    rw [ne_eq, EmbeddingLike.apply_eq_iff_eq, Sum.inl.injEq]
+    assumption'
+
+lemma S'_σpos_block (A : Matrix (Fin M) (Fin N) 𝕂) :
+    A.svdS'.toBlocks₁₁ = A.svdμ' := by
+  unfold toBlocks₁₁ svdμ' svdS'
+  simp only [reindex_apply, submatrix_apply, ne_eq, EmbeddingLike.apply_eq_iff_eq, Sum.inl.injEq,
+    submatrix_diagonal_equiv]
+  funext i j
+  by_cases h: i=j
+  · simp_rw [h]
+    unfold eigenRowEquiv finRankEquivEigsMulConjTranspose
+      Equiv.sumCongr Sum.map
+    simp only [ne_eq, Equiv.symm_trans_apply, Equiv.symm_symm, Equiv.coe_fn_symm_mk,
+      Sum.elim_inl, Equiv.sumCompl_apply_inl, of_apply,
+      diagonal_apply_eq, Function.comp_apply]
+  · rw [diagonal_apply_ne, of_apply, diagonal_apply_ne]
+    rw [ne_eq, EmbeddingLike.apply_eq_iff_eq, Sum.inl.injEq]
+    assumption'
+
+lemma S_block (A : Matrix (Fin M) (Fin N) 𝕂) :
+    (reindex
+      (eigenColumnEquiv A)
+      (eigenColumnEquiv A))
+        ( diagonal ( (isHermitian_transpose_mul_self A).eigenvalues)) =
+          fromBlocks A.svdμ 0 0 0 := by
+  let hz := S_zero_blocks A
+  rw [← svdS, ← fromBlocks_toBlocks (A.svdS), ← S_σpos_block, hz.1, hz.2.1, hz.2.2]
+
+lemma S'_block (A : Matrix (Fin M) (Fin N) 𝕂) :
+    (reindex
+      (eigenRowEquiv A)
+      (eigenRowEquiv A))
+        (diagonal ( (isHermitian_mul_conjTranspose_self A).eigenvalues)) =
+          fromBlocks A.svdμ' 0 0 0 := by
+  let hz := S'_zero_blocks A
+  rw [← svdS', ← fromBlocks_toBlocks (A.svdS'), ← S'_σpos_block, hz.1, hz.2.1, hz.2.2]
+
+lemma V_columns (A : Matrix (Fin M) (Fin N) 𝕂) :
+    (reindex (Equiv.refl (Fin N)) (eigenColumnEquiv A))
+      (isHermitian_transpose_mul_self A).eigenvectorMatrix =
+      fromColumns A.svdV₁ A.svdV₂ := by
+  rw [reindex_apply, fromColumns, svdV₁, svdV₂, toColumns₁, toColumns₂]
+  funext i j
+  cases' j with j j <;>
+  simp only [Equiv.refl_symm, Equiv.coe_refl, submatrix_apply, id_eq,
+    reindex_apply, of_apply, Sum.elim_inl, Sum.elim_inr]
+
+lemma reduced_spectral_theorem (A : Matrix (Fin M) (Fin N) 𝕂) :
```

Comment:
We indicate theorems with a popularly known name by writing their name in bold above. So something like:
```suggestion
/-- **Reduced spectral theorem**, right eigenvector version. -/
lemma V₁_mul_μ_mul_V₁_conjTranspose (A : Matrix (Fin M) (Fin N) 𝕂) :
```